RecoLocalTracker/SiPixelRecHits/interface/pixelCPEforDevice.h

```diff
@@ -234,17 +234,30 @@ namespace pixelCPEforDevice {
     if (cp.minCol[ic] == 0 || cp.maxCol[ic] == uint32_t(detParams.nCols - 1))
       cp.ysize[ic] = -cp.ysize[ic];
 
+    /* original code, kept for documentation
+
     // apply the lorentz offset correction
     float xoff = 0.5f * float(detParams.nRows) * comParams.thePitchX;
     float yoff = 0.5f * float(detParams.nCols) * comParams.thePitchY;
 
-    //correction for bigpixels for phase1
+    // correction for big pixels for Phase 1
     xoff = xoff + TrackerTraits::bigPixXCorrection * comParams.thePitchX;
     yoff = yoff + TrackerTraits::bigPixYCorrection * comParams.thePitchY;
 
     // apply the lorentz offset correction
     auto xPos = detParams.shiftX + (comParams.thePitchX * 0.5f * float(mx)) - xoff;
     auto yPos = detParams.shiftY + (comParams.thePitchY * 0.5f * float(my)) - yoff;
+    */
+
+    /* code rearranged to simplify redundant operations and use explicit FMA instructions to improve reproducibility */
+
+    float xPos = std::fmaf(0.5f * ((float)mx - (float)detParams.nRows) - TrackerTraits::bigPixXCorrection,
+                           comParams.thePitchX,
+                           detParams.shiftX);
```

Comment:
to counter the comment above, I feel like the new code is a bit more readable ;)
There is an extra effort to parse from `fmaf` to what it is in plain arithmetic operations

Reply:
Would this be a good description of the code ?
```c++
// compute the position relative to the center of the module, taking into account the corrections for the Phase 1 big pixels, the pitch, and the Lorentz shift
```